repair_picking_after_done/models/stock_move.py

```diff
@@ -0,0 +1,96 @@
+from odoo import _, models
+from odoo.exceptions import UserError
+from odoo.tools import float_is_zero, float_round
+from odoo.tools.float_utils import float_compare
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _split(self, qty, restrict_partner_id=False):
+        """Splits `self` quantity and return values
+        for a new moves to be created afterwards
+
+        This method is based on the core Odoo `stock.move`
+        `_split` method with the following customization:
+        - Allows splitting stock moves related to repair
+        orders that are marked as "done", which is prevented
+        by the core Odoo logic.
+        - This change enables the creation of backorders for
+        these split stock moves when the associated repair is completed.
+
+        Note: This customization overrides logic in both the `stock`
+        and `repair` modules to ensure that backorders can be created
+        for completed repairs, while adhering to the business rules of both modules.
+
+        :param qty: float. quantity to split (given in product UoM)
+        :param restrict_partner_id: optional partner that can be given in order
+        to force the new move to restrict its choice of quants to the ones
+        belonging to this partner.
+        :returns: list of dict. stock move values"""
+        self.ensure_one()
+
+        # Custom logic: Prevent splitting stock moves tied to incomplete repairs
+        if self.repair_id and self.repair_id.state != "done":
+            return []
+
+        # --- Core logic begins here ---
```

Comment:
I don't like this overwrite either. I see a couple of options:

A. Do a propoer hook changing the behavior of the `repair`part, example: https://github.com/OCA/stock-logistics-workflow/blob/16.0/stock_account_product_run_fifo_hook/hooks.py

B. Use a different m2o instead of repair_id to differenciate moves that are not estrictly related to the execution of the repair itself. Same idea as in MOs that have different stock moves fields for final product (`move_finished_ids`) and components (`move_raw_ids`) because they are separate concepts that you don't want to mix. Maybe in this case we could come up with a `after_repair_id` field for the relation.

What do you think?

Reply:
Thank you again for your feedback @LoisRForgeFlow!

I went with the hook option you suggested, based on the examples you shared.

Could you please take a look and confirm if everything looks correct? Let me know if any adjustments are needed. Thanks again!